docs/sphinx/examples/qec/python/custom_repetition_code_fine_grain_noise.py

```diff
@@ -0,0 +1,187 @@
+            qec.operation.x: x_logical,
+            qec.operation.stabilizer_round: stabilizer_round
+        }
+
+    def __make_stabilizers_strings(self):
+        # Create ZZ stabilizer strings: e.g., "ZZI", "IZZ", etc.
+        d = self.distance
+        return ['I' * i + 'ZZ' + 'I' * (d - i - 2) for i in range(d - 1)]
+
+    def __make_pauli_observables(self):
+        # Logical Z is Z on all data qubits
+        d = self.distance
+        return ["Z" * d]
+
+    # --- Required methods for the QEC backend ---
+
+    def get_num_data_qubits(self):
+        return self.distance
+
+    def get_num_ancilla_x_qubits(self):
+        return 0  # Not needed for Z-type repetition code
+
+    def get_num_ancilla_z_qubits(self):
+        return self.distance - 1
+
+    def get_num_ancilla_qubits(self):
+        return self.get_num_ancilla_z_qubits() + self.get_num_ancilla_x_qubits()
+
+    def get_num_x_stabilizers(self):
+        return 0
+
+    def get_num_z_stabilizers(self):
+        return self.distance - 1
+
+
+# Instantiate the custom repetition code
+my_repetition_code = qec.get_code("custom_repetition_code", distance=distance)
+print(f"\n Created custom repetition code with distance {distance}.")
+
+# Show available codes (doesn't include Python-defined custom codes)
+available_codes = qec.get_available_codes()
+print("\n  Available built-in QEC codes:", available_codes)
+
+all_codes = qec.get_all_available_codes()
+print("\n  Available QEC codes both in the library and in Python:", all_codes)
+
+# Let's check some propreties to verify that code is correctly created
+# Display the code's stabilizer generators
+stabilizers = my_repetition_code.get_stabilizers()
+print(f"\n The code has {len(stabilizers)} stabilizers:")
+for s in stabilizers:
+    print(" ", s)
+
+logical_single_round = my_repetition_code.get_observables_z()
+
+# Define and register a noise model
+noise_model = cudaq.NoiseModel()
+p = 0.01  # depolarizing noise strength
+
+noise_model.add_all_qubit_channel("x", cudaq.Depolarization2(p), 1)
+
+# Set initial logical state to |0⟩
+statePrep = qec.operation.prep0
+
+# Generate a full detector error model (DEM)
+print("\n Generating detector error model...")
+dem_rep = qec.dem_from_memory_circuit(my_repetition_code, statePrep, nRounds,
+                                      noise_model)
+
+# Extract H matrix (syndrome parity checks)
+H_pcm = dem_rep.detector_error_matrix
+
+# Extract observable flips matrix (maps physical errors to logical flips)
+Lz_observables_flips_matrix = dem_rep.observables_flips_matrix
+
+# Sample noisy executions of the full memory circuit
+print("\n Sampling noisy memory circuit executions...")
+syndromes, data = qec.sample_memory_circuit(my_repetition_code, statePrep,
+                                            nShots, nRounds, noise_model)
+
+# Reshape syndromes to flatten rounds per shot
+syndromes = syndromes.reshape((nShots, nRounds, -1))
+syndromes = syndromes.reshape((nShots, -1))
```

Comment:
I think the first line is unnecessary.
```suggestion
syndromes = syndromes.reshape((nShots, -1))
```